### PR TITLE
[docs] Add missing default `loading` prop value

### PR DESCRIPTION
**Pull request by oliviertassinari**
## Before

SCR-20240623-mmuz

https://mui.com/x/api/data-grid/data-grid/#data-grid-prop-loading

## After

https://deploy-preview-13604--material-ui-x.netlify.app/x/api/data-grid/data-grid/#data-grid-prop-loading

### CODE REVIEW

File: packages/x-data-grid/src/models/props/DataGridProps.ts [hunk 1]
```diff
@@ -736,6 +736,7 @@ export interface DataGridPropsWithoutDefaultValue<R extends GridValidRowModel =
   getRowId?: GridRowIdGetter<R>;
   /**
    * If `true`, a loading overlay is displayed.
+   * @default false
```
Comment (oliviertassinari) [inline packages/x-data-grid/src/models/props/DataGridProps.ts:739]: I just had to update this row and `pnpm proptypes && pnpm docs:api` did the rest.

Comment (MBilalShafi) [reply packages/x-data-grid/src/models/props/DataGridProps.ts:739]: We don't usually add `@default` to the values in `DataGridPropsWithoutDefaultValue`. Should we also move the `loading` prop to `DataGridPropsWithDefaultValues`?

Comment (oliviertassinari) [reply packages/x-data-grid/src/models/props/DataGridProps.ts:739]: No preferences, it adds a bit of bundle size but is also clearer.

**Review by MBilalShafi — APPROVED** [review]

**Comment by oliviertassinari** [comment]
@MBilalShafi Thanks for completing the PR.

**Pull request merged**